CommunicationLayer/CommDeviceControl/InternetConnectionService.cpp

```diff
@@ -0,0 +1,93 @@
+#include <SimpleAmqpClient/SimpleAmqpClient.h>
+#include <QUdpSocket>
+#include "InternetConnectionService.h"
+#include <QDebug>
+#include "../../InfrastructureLayer/Settings/I_Settings.h"
+#include <QTextStream>
+#include <QTimer>
+namespace
+{
+    quint32 TIMEOUT_MILLISECONDS = 120000;
+    quint32 SLEEP_TIME_MILLISECONDS = 2000;
+}
+
+
+InternetConnectionService::InternetConnectionService(QString exchangeName,
+        QString ipAddress,
+        quint16 udpPort)
+    : exchangeName_(exchangeName),
+      ipAddress_(ipAddress),
+      udpPort_(udpPort)
+{
+    QObject::connect(this, SIGNAL(setupChannelSignal()), this, SLOT(setupChannel()));
+
+    qtimer_ = new QTimer(this);
+    qtimer_->setSingleShot(true);
+    connect(qtimer_, SIGNAL(timeout()), this, SLOT(setupChannel()));
+    connectToDataSource();
+}
+
+InternetConnectionService::~InternetConnectionService()
+{
+}
+
+
+void InternetConnectionService::setupChannel()
+{
+    qWarning() << "UdpMessageForwarder: Attempting to connect" << ipAddress_ << udpPort_;
+
+    try
+    {
+        channel_ = AmqpClient::Channel::Create(ipAddress_.toStdString(), udpPort_);
+    }
+    catch (std::exception&)
```

Comment:
Dont' use a catch all, this is a common code smell. The reason it's bad is because you don't know actually know what you're catching, meaning you don't actually know how to handle the errors. Some of these exceptions may mean more than "it didn't work".